x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunnerTests.java

```diff
@@ -218,10 +220,207 @@ public void testRunnerMultiSource() throws Exception {
         Map<?, ?> properties = (Map<?, ?>) mapping.get("properties");
         assertNotNull(properties);
         assertThat(properties.size(), is(equalTo(1)));
-        Map<?, ?> field1 = (Map<?, ?>) properties.get("field1");
-        assertNotNull(field1);
-        assertThat(field1.get("type"), is(equalTo("keyword")));
-        assertThat(field1.get("doc_values"), is(false));
+        Map<?, ?> keyfield = (Map<?, ?>) properties.get("key");
+        assertNotNull(keyfield);
+        assertThat(keyfield.get("type"), is(equalTo("keyword")));
+        assertThat(keyfield.get("doc_values"), is(false));
+
+        // Validate document structure
+        SearchResponse enrichSearchResponse = client().search(
+            new SearchRequest(".enrich-test1")
+                .source(SearchSourceBuilder.searchSource()
+                    .query(QueryBuilders.matchAllQuery()))).actionGet();
+        assertThat(enrichSearchResponse.getHits().getTotalHits().value, equalTo(3L));
+        Map<String, Object> enrichDocument = enrichSearchResponse.getHits().iterator().next().getSourceAsMap();
+        assertNotNull(enrichDocument);
+        assertThat(enrichDocument.size(), is(equalTo(5)));
+        assertThat(enrichDocument.get("key"), is(equalTo("key0")));
+        assertThat(enrichDocument.get("field1"), is(equalTo("value1")));
+        assertThat(enrichDocument.get("field2"), is(equalTo(2)));
+        assertThat(enrichDocument.get("field5"), is(equalTo("value5")));
+
+        // Validate segments
+        validateSegments(createdEnrichIndex, 3);
+
+        // Validate Index is read only
+        ensureEnrichIndexIsReadOnly(createdEnrichIndex);
+    }
+
+    public void testRunnerMultiSourceDocIdCollisions() throws Exception {
+        String baseSourceName = "source-index-";
+        int numberOfSourceIndices = 3;
+        String collidingDocId = randomAlphaOfLength(10);
+        for (int idx = 0; idx < numberOfSourceIndices; idx++) {
+            final String sourceIndex = baseSourceName + idx;
+            IndexResponse indexRequest = client().index(new IndexRequest()
+                .index(sourceIndex)
+                .id(collidingDocId)
+                .source(
```

Comment:
I think we should set specific _routing value here too and verify that in the .enrich index it is no longer there?

Normally if a _routing gets specified it gets indexed into a `_routing` field that is also queryable, so
testing that it is queryable in the source index and not in the enrich index should be sufficient.